apps/mobile/src/features/ImportReadOnly/ScanQrAccount.container.tsx

```diff
@@ -0,0 +1,97 @@
+import { Camera, Code, useCameraPermission } from 'react-native-vision-camera'
+import { Text, View } from 'tamagui'
+import React, { useCallback, useEffect, useState } from 'react'
+import { useFocusEffect, useRouter } from 'expo-router'
+import { SafeButton } from '@/src/components/SafeButton'
+
+import { SafeFontIcon } from '@/src/components/SafeFontIcon'
+import { QrCamera } from '@/src/components/Camera'
+import { parsePrefixedAddress } from '@safe-global/utils/addresses'
+import { isValidAddress } from '@safe-global/utils/validation'
+import { ToastViewport, useToastController } from '@tamagui/toast'
+
+const toastForValueShown: Record<string, boolean> = {}
+export const ScanQrAccountContainer = () => {
+  const router = useRouter()
+  const [isCameraActive, setIsCameraActive] = useState(true)
+  const toast = useToastController()
+
+  const permission = Camera.getCameraPermissionStatus()
+  const { hasPermission, requestPermission } = useCameraPermission()
+
+  useEffect(() => {
+    if (hasPermission === false && permission === 'not-determined') {
+      requestPermission()
+    }
+  }, [permission, hasPermission, requestPermission])
+
+  useFocusEffect(
+    // Callback should be wrapped in `React.useCallback` to avoid running the effect too often.
+    useCallback(() => {
+      // Invoked whenever the route is focused.
+      setIsCameraActive(true)
+
+      // Return function is invoked whenever the route gets out of focus.
+      return () => {
+        setIsCameraActive(false)
+      }
+    }, []),
+  )
+
+  const onScan = useCallback(
+    (codes: Code[]) => {
+      if (codes.length > 0 && isCameraActive) {
```

Comment:
We could adopt early returns in order to have 1 less line ident level
```suggestion
      if (!codes.length || !isCameraActive) {
       return;
       }
```